src/common/transformations/src/transformations/common_optimizations/division_by_zero_fp16_resolver.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include "transformations/common_optimizations/division_by_zero_fp16_resolver.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <openvino/opsets/opset8.hpp>
+#include "ngraph/rt_info.hpp"
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+
+NGRAPH_RTTI_DEFINITION(ov::pass::DivisionByZeroFP16Resolver, "DivisionByZeroFP16Resolver", 0);
+
+constexpr float normalized_fp16_min = 6.103515625e-05f;  // fp16 minimal normalized  value
+
+using namespace ov;
+
+ov::pass::DivisionByZeroFP16Resolver::DivisionByZeroFP16Resolver() {
+    MATCHER_SCOPE(DivisionByZeroFP16Resolver);
+    auto input_1 = pattern::any_input();
+    auto input_2 = pattern::any_input();
+
+    auto eps_const_pattern = pattern::wrap_type<opset8::Constant>();
+    auto max = std::make_shared<opset8::Maximum>(input_2, eps_const_pattern);
+    auto add = std::make_shared<opset8::Add>(input_2, eps_const_pattern);
+    auto max_or_add = std::make_shared<pattern::op::Or>(OutputVector{max, add});
+    auto divide = std::make_shared<opset8::Divide>(input_1, max_or_add);
+    auto pow_exp = pattern::wrap_type<opset8::Constant>();
+    auto pow_pattern = std::make_shared<opset8::Power>(max_or_add, pow_exp);
+    auto div_or_pow = std::make_shared<pattern::op::Or>(OutputVector{divide, pow_pattern});
```

Comment:
Could you explain why this exact patters are used?

Reply:
This was not the exact pattern, I had missed a couple of cases but now the list is full.
We have to catch all cases there division to zero can happen: 
1. `input_1/Maximum(input_2, eps)`
2. `input_1/Add(input_2, eps)`
3. `input_1/Sqrt(Maximum(input_2, eps))`
4. `input_1/Sqrt(Add(input_2, eps))`
5. `input_1*Pow(Maximum(input_2, eps), -z)`
6. `input_1*Pow(Add(input_2, eps), -z)`

added these patterns into comments. Pattern `5.` and `6.` can be found in `L2Norm` subgraphs. from `1.` to `4.` - at least in `MVN` sugraphs, but no only